packages/istanbul-lib-instrument/src/instrumenter.js

```diff
@@ -80,7 +80,13 @@ class Instrumenter {
         const opts = this.opts;
         const ast = babylon.parse(code, {
             allowReturnOutsideFunction: opts.autoWrap,
-            sourceType: opts.esModules ? "module" : "script"
+            sourceType: opts.esModules ? "module" : "script",
+            plugins: [
+              'asyncGenerators',
+              'dynamicImport',
+              'flow',
+              'jsx'
+            ]
```

Comment:
Sorry I havent had time to work more on the original pr. Either way I was looking at the tests there, the yaml specs and to be fair I don't know enough about how the instrumenter works to put together a test there, the whole expected thing seems cryptic, it generates a bunch of numbers encoded in json - is this documented somewhere exactly what that is? I mean, I have a hunch its lines/columns etc - but yeah not really sure. If someone else could pitch on with some idea of how to test this, that'd be awesome - adding another spec with jsx there etc.